### PR TITLE
Base POCO parsers/serializers on POCO, not IROD

### LINKED ISSUE #2917: Refactor the Poco serializers to work against Base
Change the serializers to use the new IScopedNode interface.

### PR DESCRIPTION
**Pull request by ewoutkramer**
* ⚠️  BREAKING: Based the parsers/serializers on Base instead of IReadOnlyDictionary<>.  Should not really matter, since internally, the code required the things to be subclasses of Base anyway to function correctly.
* Added extensions so subtrees van be serialized, if the underlying engine is our new POCO serializer
* Added extensions so one can serialize to/from streams, if the underlying engine is our new POCO serializer.
* ⚠️ BREAKING: Removed the obsolete FhirJsonConverter constructors (use the ForFhir extension methods on JsonSerializerOptions).
* ⚠️ BREAKING: Renamed methods on the IFhirSerializationExtensions, since they made no sense (e.g. `DeserializeReaderToJson` has become `DeserializeFromJsonReader`).

Fixes #2917.

### CODE REVIEW

File: src/Hl7.Fhir.Base/Serialization/BaseFhirJsonPocoSerializer.cs [hunk 1]
```diff
@@ -20,314 +20,310 @@
...
+    /// <summary>
+    /// Serializes the given dictionary with FHIR data into Json.
+    /// </summary>
+    public void Serialize(Base element, Utf8JsonWriter writer) =>
+        serializeInternal(element, writer, skipValue: false);
+    
+    /// <summary>
+    /// Serializes the given dictionary with FHIR data into a Json string.
+    /// </summary>
+    public string SerializeToString(Resource element)
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Base/Serialization/BaseFhirJsonPocoSerializer.cs]: Should be `Base` too to be consistent.

Comment (ewoutkramer) [reply src/Hl7.Fhir.Base/Serialization/BaseFhirJsonPocoSerializer.cs]: Done.

File: src/Hl7.Fhir.Base/Serialization/FhirXmlBuilderExtensions.cs [hunk 2]
```diff
@@ -84,8 +88,11 @@ public static async Task<string> ToXmlAsync(this ITypedElement source, FhirXmlSe
                 return await SerializationUtil.WriteXmlToStringAsync(async writer => await source.WriteToAsync(writer, settings).ConfigureAwait(false), settings?.Pretty ?? false,
                     settings?.AppendNewLine ?? false).ConfigureAwait(false);
             
-            var engine = FhirSerializationEngineFactory.Strict(ModelInspector.ForType(b.GetType()));
-            return ((PocoSerializationEngine)engine).SerializeToXml(b);
+            // Note that this code is temporary, as we the above code will be re-instated here. It's therefore
```
Comment (Kasdejong) [inline src/Hl7.Fhir.Base/Serialization/FhirXmlBuilderExtensions.cs:91]: if it is temporary, maybe we should mark it with [TemporarilyChanged]?

File: src/Hl7.Fhir.Base/Serialization/FhirXmlBuilderExtensions.cs [hunk 3]
```diff
@@ -71,10 +71,14 @@ public static async Task<string> ToXmlAsync(this ISourceNode source, FhirXmlSeri
...
         {
             if (source is not Base b)
-                return SerializationUtil.WriteXmlToString(writer => source.WriteTo(writer, settings), settings?.Pretty ?? false, settings?.AppendNewLine ?? false);
-            
-            var engine = FhirSerializationEngineFactory.Strict(ModelInspector.ForType(b.GetType()));
-            return ((PocoSerializationEngine)engine).SerializeToXml(b);
+                return SerializationUtil.WriteXmlToString(source, (s,w) => s.WriteTo(w, settings),
+                    settings?.Pretty ?? false, settings?.AppendNewLine ?? false);
+
+            // Note that this code is temporary, as we the above code will be re-instated here. It's therefore
```
Comment (Kasdejong) [inline src/Hl7.Fhir.Base/Serialization/FhirXmlBuilderExtensions.cs:77]: if it is temporary, maybe we should mark it with [TemporarilyChanged]?

**Review by Kasdejong — CHANGES_REQUESTED** [review]
I accidentally set the review to "comment" instead of "request changes", this should tag it appropriately

**Review by Kasdejong — APPROVED** [review]

**Pull request merged**